nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2485 @@
+
+    See Also
+    --------
+    nilearn.datasets.neurovault.IsNull,
+    nilearn.datasets.neurovault.NotNull,
+    nilearn.datasets.neurovault.NotEqual,
+    nilearn.datasets.neurovault.GreaterOrEqual,
+    nilearn.datasets.neurovault.GreaterThan,
+    nilearn.datasets.neurovault.LessOrEqual,
+    nilearn.datasets.neurovault.LessThan,
+    nilearn.datasets.neurovault.IsIn,
+    nilearn.datasets.neurovault.NotIn,
+    nilearn.datasets.neurovault.Contains,
+    nilearn.datasets.neurovault.Pattern.
+
+    """
+    def __init__(self, *must_not_be_contained):
+        self.must_not_be_contained_ = must_not_be_contained
+
+    def __eq__(self, other):
+        if not isinstance(other, Container):
+            return False
+        for item in self.must_not_be_contained_:
+            if item in other:
+                return False
+        return True
+
+    def __repr__(self):
+        return '{0}{1!r}'.format(
+            self.__class__.__name__, self.must_not_be_contained_)
+
+
+class Pattern(_SpecialValue):
+    """Special value used to filter terms.
+
+    An instance of this class is constructed with
+
+    `Pattern(pattern[, flags])`. It will allways be equal to, and only
+    to, any value for which ``re.match(pattern, value, flags)`` is
+    ``True``.
+
+    Parameters
+    ----------
+    pattern : str
+        The pattern to try to match to candidates.
+
+    flags : int, optional (default=0)
+        Value for ``re.match`` `flags` parameter,
+        e.g. ``re.IGNORECASE``. The default (0), is the default value
+        used by ``re.match``.
+
+    See Also
+    --------
+    nilearn.datasets.neurovault.IsNull,
+    nilearn.datasets.neurovault.NotNull,
+    nilearn.datasets.neurovault.NotEqual,
+    nilearn.datasets.neurovault.GreaterOrEqual,
+    nilearn.datasets.neurovault.GreaterThan,
+    nilearn.datasets.neurovault.LessOrEqual,
+    nilearn.datasets.neurovault.LessThan,
+    nilearn.datasets.neurovault.IsIn,
+    nilearn.datasets.neurovault.NotIn,
+    nilearn.datasets.neurovault.Contains,
+    nilearn.datasets.neurovault.NotContains.
+
+    Documentation for standard library ``re`` module.
+
+    Examples
+    --------
+    >>> from nilearn.datasets.neurovault import Pattern
+    >>> pattern = Pattern(r'[0-9akqj]{5}$')
+    >>> 'ak05q' == pattern
+    True
+    >>> 'ak05e' == pattern
+    False
+
+    """
+    def __init__(self, pattern, flags=0):
+        # Don't use re.compile because compiled patterns
+        # can't be deepcopied.
+        self.pattern_ = pattern
+        self.flags_ = flags
+
+    def __eq__(self, other):
+        if not isinstance(other, _basestring) or re.match(
+                self.pattern_, other, self.flags_) is None:
+            return False
+        return True
+
+    def __repr__(self):
+        return '{0}(pattern={1!r}, flags={2})'.format(
+            self.__class__.__name__, self.pattern_, self.flags_)
+
+
+def _empty_filter(arg):
+    """Place holder for a filter which always returns True."""
```

Comment:
It would have been nice if you specify for which argument this default function is used. To understand where it is used makes easy.